packages/core/src/helpers.ts

```diff
@@ -152,9 +152,8 @@ export default class Helpers implements Resolver {
         }
     }
 
-    // TODO: specialize fetch to retrieve just thing descriptions
-    public fetch(uri: string): Promise<unknown> {
-        return new Promise<unknown>((resolve, reject) => {
+    public fetch(uri: string): Promise<ThingDescription> {
+        return new Promise<ThingDescription>((resolve, reject) => {
```

Comment:
`unknown` here is correct because the method does not perform any validation on the returned json object. We could have fetched a JSON like `"hello I am not a TD"`. Either we do the validation in this method or we return unknown and let the user do the validation or the forced casting. returning ThingDescription is wrong.

Reply:
Mhh, if you take a look at all the examples WotHelpers.fetch() expect as result a ThingDescription.
Not sure how to deal with it properly... shall we "intruduce" the checking or leave it as is?

Reply:
It depends on how we consider the fetch function. If it is actually a `fetchThingDescription` (as all the examples are using it) then we should do the validation in the function and change the return type as you did.

Reply:
An alternative could be to create yet another helper function `fetchTD` that uses `fetch` as is but does in addition the validation. What do you think?

Reply:
IDK, it feels like we don't need a generic `fetch` function... So I would simply refactor the existing one (even if technically is a change in the API).

Reply:
Two comments:
1. I am still unsure whether we should change the existing `fetch` call to do something in advance. I still think it is better to create a dedicated `fetchTD` call. @JKRhb @egekorkan any comments/opinions
2. We do not have a nice function to validate a TD (at least not on the basis of the schema). Hence we need to add this... similarly as we do for ThingModels

Reply:
yes, plus one for point 2 definitely. For point 1 let's see what others think.

Reply:
Mhh, my personal opinion is that the helpers functions are independent from a discovery API. They are just helpers to "fetch" data. Hence my proposal would be to
* leave `fetch` as is returning `unknown`
* introduce `fetchTD` that at the moment simply casts  `unknown` to `ThingDescription` (in a follow-up PR we can decide which kind of validation can/should be implemented)

Anyhow, if this is too much change the safer way is to go with the proposal from @relu91

Reply:
To make progress with this PR I think it is best to go with the proposal from @relu91.
Hence I created https://github.com/eclipse-thingweb/node-wot/issues/1055

Note: Updates to this PR will follow...